Source/WTF/wtf/text/WYHasher.h

```diff
@@ -0,0 +1,264 @@
+    {
+        wymum(&A, &B);
+        return A ^ B;
+    }
+
+#define enable_is_constant_evaluated 0
+
+    template<typename T>
+    struct Reader16Bit {
+        ALWAYS_INLINE static constexpr uint64_t wyr2(const T* p)
+        {
+            ASSERT(sizeof(T) == 2);
+            uint64_t firstByte = static_cast<uint8_t>(p[0]);
+            uint64_t secondByte = static_cast<uint8_t>(p[0] >> 8);
+            return (firstByte << 16) | (secondByte << 8) | secondByte;
+        }
+
+        ALWAYS_INLINE static constexpr uint64_t wyr4(const T* p)
+        {
+            ASSERT(sizeof(T) == 2);
+            if (std::is_constant_evaluated() || enable_is_constant_evaluated) {
+                return static_cast<uint32_t>(p[0])
+                    | (static_cast<uint32_t>(p[1]) << 16);
+            }
+            uint32_t v;
+            memcpy(&v, p, 4);
+            return v;
+        }
+
+        ALWAYS_INLINE static constexpr uint64_t wyr8(const T* p)
+        {
+            ASSERT(sizeof(T) == 2);
+            if (std::is_constant_evaluated() || enable_is_constant_evaluated) {
+                return static_cast<uint64_t>(p[0])
+                    | (static_cast<uint64_t>(p[1]) << 16)
+                    | (static_cast<uint64_t>(p[2]) << 32)
+                    | (static_cast<uint64_t>(p[3]) << 48);
+            }
+            uint64_t v;
+            memcpy(&v, p, 8);
+            return v;
+    }
+    };
+
+    template<typename T>
+    struct Reader8Bit {
+        ALWAYS_INLINE static constexpr uint64_t wyr2(const T* p)
+        {
+            ASSERT(sizeof(T) == 1);
+            return static_cast<uint64_t>(p[0]) << 16;
+        }
+
+        ALWAYS_INLINE static constexpr uint64_t wyr4(const T* p)
+        {
+            ASSERT(sizeof(T) == 1);
+            if (std::is_constant_evaluated() || enable_is_constant_evaluated)
+                return static_cast<uint32_t>(p[0]) | (static_cast<uint32_t>(p[1]) << 16);
+            uint32_t v;
+            memcpy(&v, p, 2);
+            return (v | (v << 8)) & 0x00ff00ffUL;
+        }
+
+        ALWAYS_INLINE static constexpr uint64_t wyr8(const T* p)
+        {
+            ASSERT(sizeof(T) == 1);
+            if (std::is_constant_evaluated() || enable_is_constant_evaluated) {
+                return static_cast<uint64_t>(p[0])
+                    | (static_cast<uint64_t>(p[1]) << 16)
+                    | (static_cast<uint64_t>(p[2]) << 32)
+                    | (static_cast<uint64_t>(p[3]) << 48);
+            }
+            uint64_t v;
+            memcpy(&v, p, 4);
+            v = (v | (v << 16)) & 0x0000ffff0000ffffULL;
+            return (v | (v << 8)) & 0x00ff00ff00ff00ffULL;
+        }
+    };
+
+    template<typename T, typename Reader>
+    static constexpr unsigned hash(const T* p, const unsigned characterCount)
+    {
+        auto wyr2 = Reader::wyr2;
+        auto wyr4 = Reader::wyr4;
+        auto wyr8 = Reader::wyr8;
+
+        const unsigned byteCount = characterCount << 1;
```

Comment:
Let's use `uint64_t` for byteCount to avoid accidental overflow.

Reply:
Do it, `const uint64_t byteCount = static_cast<uint64_t>(characterCount) << 1;`.